core/app/models/spree/wallet_payment_source.rb

```diff
@@ -1,19 +1,31 @@
 # frozen_string_literal: true
 
-class Spree::WalletPaymentSource < ActiveRecord::Base
-  belongs_to :user, class_name: Spree::UserClassHandle.new, foreign_key: 'user_id', inverse_of: :wallet_payment_sources
-  belongs_to :payment_source, polymorphic: true, inverse_of: :wallet_payment_sources
+module Spree
+  class WalletPaymentSource < Spree::Base
+    belongs_to :user, class_name: Spree::UserClassHandle.new, foreign_key: 'user_id', inverse_of: :wallet_payment_sources
+    belongs_to :payment_source, polymorphic: true, inverse_of: :wallet_payment_sources
 
-  validates_presence_of :user
-  validates_presence_of :payment_source
+    validates_presence_of :user
+    validates_presence_of :payment_source
+    validates :user_id, uniqueness: { scope: [:payment_source_type, :payment_source_id] }
```

Comment:
It may be worth adding a custom message here. I found the default message on this is going to be `User id has already been taken`. That message doesn't really denote that the violation is due to the tuple key

Reply:
Also, can we please document why this change is needed in this PR?